modules/nw-cudn-cr.adoc

```diff
@@ -0,0 +1,169 @@
+====
+
+.Prerequisites
+
+* You have logged in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Optional: For a `ClusterUserDefinedNetwork` CR that uses a primary network, create a namespace with the `k8s.ovn.org/primary-user-defined-network` label by entering the following command:
++
+[source,yaml]
+----
+$ cat << EOF | oc apply -f -
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: <cudn_namespace_name>      
+  labels:
+    k8s.ovn.org/primary-user-defined-network: ""
+EOF
+----
+
+. Create a request for either a `Layer2` or `Layer3` topology type cluster-wide user-defined network:
+
+.. Create a YAML file, such as `cluster-layer-two-udn.yaml`, to define your request for a `Layer2` topology as in the following example:
++
+[source, yaml]
+----
+apiVersion: k8s.ovn.org/v1
+kind: ClusterUserDefinedNetwork
+metadata:
+  name: <cudn_name> # <1>
+spec:
+  namespaceSelector: # <2>
+    matchLabels: # <3>
+    - "<example_namespace_one>":"" # <4>
+    - "<example_namespace_two>":"" # <4>
+  network: # <5>
+    topology: Layer2 # <6>
+    layer2: # <7>
+      role: Primary # <8>
+      subnets:
+        - "2001:db8::/64"
+        - "10.100.0.0/16" # <9>
+----
+<1> Name of your `ClusterUserDefinedNetwork` custom resource.
+<2> A label query over the set of namespaces that the cluster UDN applies to. Uses the standard Kubernetes `MatchLabel` selector. Must not point to `default` or `openshift-*` namespaces.
+<3> Uses the `matchLabels` selector type, where terms are evaluated with an `AND` relationship. 
+<4> Because the `matchLabels` selector type is used, provisions namespaces matching both `<example_namespace_one>` _and_ `<example_namespace_two>`.
+<5> Describes the network configuration.
+<6> The `topology` field describes the network configuration; accepted values are `Layer2` and `Layer3`. Specifying a `Layer2` topology type creates one logical switch that is shared by all nodes.
+<7> This field specifies the topology configuration. It can be `layer2` or `layer3`.
+<8> Specifies `Primary` or `Secondary`. `Primary` is the only `role` specification supported in {product-version}.
+<9> For `Layer2` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+* `Layer2` subnets can be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+* The `Layer2` `subnets` field is mandatory when the `ipamLifecycle` field is specified.
```

Comment:
so .... I'm torn.

So far, we've hid this attribute. IMHO we either present it before, or ignore it altogether. I'd vote for omitting it.

Then, the API was updated - this is now `ipam.lifecycle`.

Reply:
I have a separate PR for that here: https://github.com/openshift/openshift-docs/pull/88231, however I will remove it for this PR.